napari/layers/shapes/_shape_list.py

```diff
@@ -1113,17 +1121,26 @@ def inside(self, coord):
             Index of shape if any that is at the coordinates. Returns `None`
             if no shape is found.
         """
-        triangles = self._mesh.vertices[self._mesh.displayed_triangles]
-        indices = inside_triangles(triangles - coord)
-        shapes = self._mesh.displayed_triangles_index[indices, 0]
-
-        if len(shapes) == 0:
+        if not self.shapes:
+            return None
+        bounding_boxes = self._bounding_boxes
+        inside = np.all(
+            (bounding_boxes[0] <= coord) * (bounding_boxes[1] >= coord),
+            axis=1,
+        )
+        inside_indices = np.nonzero(inside)[0]
+        if not inside_indices.size:
+            return None
+        try:
+            return next(
+                i
+                for i in inside_indices
+                if np.any(
+                    inside_triangles(self.shapes[i]._all_triangles() - coord)
+                )
+            )
+        except StopIteration:
             return None
-
-        z_list = self._z_order.tolist()
-        order_indices = np.array([z_list.index(m) for m in shapes])
-        ordered_shapes = shapes[np.argsort(order_indices)]
-        return ordered_shapes[0]
```

Comment:
Looking at this deleted code, I notice that you no longer sort the shapes by z_order. Are we sure that doesn't matter? (ie are the shapes in z_order anyway?)

Reply:
I do not understand this part of the code. It looks obsolete now, to me. It comes from #100 PR